### DISCUSSION TITLE
Drop request by logic

### DISCUSSION THREAD

**Discussion by ScrimForever** [fastapi/fastapi#11647] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.requests import Request
from fastapi.background import BackgroundTasks


app = FastAPI()

@app.post("/")
async def root(request: Request):
    del request
```

### Description

How can I drop a packet that was received at the endpoint without sending a response?
I want drop requests that is not usefull 

### Operating System

Linux, Windows, Other

### FastAPI Version

0.111.0

### Pydantic Version

2.7.2

### Python Version

3.12

**Comment by mattmess1221** [comment, marked as answer]
I don't believe uvicorn or the asgi spec lets you just drop connections. Raising exceptions or returning non-responses just makes uvicorn respond with a 500 response. The closest you can get is awaiting an Event that is never set to make the client time out, but then that task is left running forever.

If this is a production issue, you may want to implement this at the firewall or reverse proxy level so none of those unwanted requests even reach your FastAPI app. Nginx will drop connections [when you return non-standard 444](https://nginx.org/en/docs/http/request_processing.html#how_to_prevent_undefined_server_names).

**Reply by ScrimForever** [reply]
The problem is that I don't manage the firewall. Today we put a 5-minute sleep and the connection is dropped due to timeout, but the session remains as an await during this period. 
I need to do something so as not to leave a “connection” open in this request that will be dropped.

**Reply by mattmess1221** [reply]
Have you tried adding another layer between you and the firewall? Installing nginx and using proxy_pass on localhost is a common task.
